packages/react-dom-bindings/src/client/ReactDOMHostConfig.js

```diff
@@ -1600,6 +1609,19 @@ export function resolveSingletonInstance(
       }
       return body;
     }
+    case 'title': {
+      if (!ownerDocument.head) {
+        throw new Error(
+          'React expected a <head> element (document.head) to exist in the Document but one was' +
+            ' not found. React never removes the head for any Document it renders into so' +
+            ' the cause is likely in some other script running on this page.',
+        );
+      }
+      // This will initialize a title element if it didn't already exist
+      ownerDocument.title = ownerDocument.title;
```

Comment:
this may be too cute and also, mutates more than necessary. not sure if browsers are smart and noop when not changed. either way I could rewrite to be more conventional about constructing a title if missing. unlike the other singletons we cannot expect the document to always contain one